src/skvhttpclient/src/skvhttp/dto/SKVRecord.h

```diff
@@ -210,29 +210,30 @@ class SKVRecordBuilder {
 public:
     // The record must be serialized in order. Schema will be enforced
     template <typename T>
-    void serializeNext(T field) {
-        if (isNan<T>(field)) {
+    void serializeNext(T&& field) {
+        if (isNan(field)) {
             throw NaNError("NaN type in serialization");
         }
 
-        FieldType ft = TToFieldType<T>();
+        using Tnr = typename std::remove_reference<T>::type;
+        FieldType ft = TToFieldType<Tnr>();
         if (_record.fieldCursor >= _record.schema->fields.size() || ft != _record.schema->fields[_record.fieldCursor].type) {
             throw TypeMismatchException("Schema not followed in record serialization");
         }
 
         for (size_t i = 0; i < _record.schema->partitionKeyFields.size(); ++i) {
             if (_record.schema->partitionKeyFields[i] == _record.fieldCursor) {
-                _record.partitionKeys[i] = FieldToKeyString<T>(field);
+                _record.partitionKeys[i] = FieldToKeyString(field);
```

Comment:
Is `SKVRecord::partitionKeys` and `rangeKeys` used anywhere in `skvhttp` ? I see fields are deserialized from `MpackReader`, not from here.

Reply:
`getKey` seems to use it.